homework2/hw3.py

```diff
@@ -0,0 +1,27 @@
+"""
+Write a function that takes K lists as arguments and returns all possible
+lists of K items where the first element is from the first list,
+the second is from the second and so one.
+
+You may assume that that every list contain at least one element
+
+Example:
+
+assert combinations([1, 2], [3, 4]) == [
+    [1, 3],
+    [1, 4],
+    [2, 3],
+    [2, 4],
+]
+"""
+import itertools
+from typing import Any, List
+
+
+def combinations(*args: List[Any]) -> List[List]:
+    answer = []
+
+    for i in itertools.product(*args):
+        answer.append(list(i))
+
+    return answer
```

Comment:
Could you make it without creating an `answer` list upfront?

Reply:
You are correct I can use comprehension here `[list(i) for i in itertools.product(*args)]`